dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/merge/indicator/IndicatorMergeProcessor.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2004-2024, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.merge.indicator;
+
+import lombok.RequiredArgsConstructor;
+import org.hisp.dhis.merge.MergeProcessor;
+import org.hisp.dhis.merge.MergeService;
+import org.springframework.stereotype.Component;
+
+/**
+ * Implementation of {@link MergeProcessor} that currently only uses its default method.
+ *
+ * @author david mackessy
+ */
+@Component
+@RequiredArgsConstructor
+public class IndicatorMergeProcessor implements MergeProcessor {
+  private final MergeService indicatorMergeService;
+
+  @Override
+  public MergeService getMergeService() {
+    return indicatorMergeService;
```

Comment:
I feel like there now is too much surface API concerned with merge. We have service, handler, processor. 
What is a bit inconsistent with the terms as we use it normally is that there is one `MergeService` per merge type which is resolved. This creates the need to keep track of the correct implementation for the particular type. Wouldn't it be simpler to make that part of the implementation become private by adding the `MergeType` to the `MergeParams`  so that the same `MergeService` (to the outside) handles any merge request. Inside we could still have the different classes/components be responsible for one type each, just that this code organisation is not exposed to the caller making it harder to understand how to call this correctly? Makes sense?
And that with that change I hope that maybe one of the handler or processor abstractions also dissolves. But we talked about this and this was something like the glue between controller and service call chains. So if we still need it that is fine.

Reply:
we chatted about this and are happy with the current implementation, after walking through a few things.